packages/mobile/src/identity/securityCode.ts

```diff
@@ -0,0 +1,119 @@
+import {
+  ActionableAttestation,
+  AttestationsWrapper,
+} from '@celo/contractkit/lib/wrappers/Attestations'
+import { PhoneNumberHashDetails } from '@celo/contractkit/src/identity/odis/phone-number-identifier'
+import { GetAttestationRequest } from '@celo/utils/lib/io'
+import { Address } from '@celo/utils/src/address'
+import BigNumber from 'bignumber.js'
+import Logger from 'src/utils/Logger'
+
+const TAG = 'identity/securityCode'
+
+const cache: { [key: string]: string } = {}
+
+function hashAddressToSingleDigit(address: Address): number {
+  return new BigNumber(address.toLowerCase()).modulo(10).toNumber()
+}
+
+export function getSecurityCodePrefix(issuerAddress: Address) {
+  if (!cache[issuerAddress]) {
+    cache[issuerAddress] = `${hashAddressToSingleDigit(issuerAddress)}`
+  }
+  return cache[issuerAddress]
+}
+
+export function extractSecurityCodeWithPrefix(message: string) {
+  const matches = message.match('\\s(\\d{8})\\s')
+  if (matches && matches.length === 2) {
+    return matches[1]
+  }
+  return null
+}
+
+// We look for the case, where all the promises fail or the first one to succeed.
+// Promise.all looks for all the values or the first error, so we can switch roles
+// for reject and resolve to achieve what we need
+async function raceUntilSuccess<T>(promises: Array<Promise<T>>) {
+  try {
+    const errors = await Promise.all(
```

Comment:
We can use `Promise.any` and not have to switch the definitions

Reply:
Unfortunately it's not a part of https://github.com/microsoft/TypeScript/blob/master/lib/lib.es2015.promise.d.ts